src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -723,13 +723,16 @@ InferenceProfiler::ProfileHelper(
     measurement_perf_status.request_rate = experiment_perf_status.request_rate;
     RETURN_IF_ERROR(manager_->CheckHealth());
 
+    MeasureConfig measure_config;
     if (measurement_mode_ == MeasurementMode::TIME_WINDOWS) {
-      error.push(
-          Measure(measurement_perf_status, measurement_window_ms_, false));
+      measure_config.measurement_window = measurement_window_ms_;
+      measure_config.is_count_based = false;
     } else {
-      error.push(
-          Measure(measurement_perf_status, measurement_request_count_, true));
+      measure_config.measurement_window = measurement_request_count_;
+      measure_config.is_count_based = true;
     }
+    measure_config.clamp_window = (request_count != 0);
```

Comment:
How does request_count tell us if we need to clamp the window?

Reply:
Request count was the new feature added recently where someone can ask to run for exactly X requests. By default it is 0, which means the feature is off and we should run normally. We only want to clamp the window for this new feature, because we aren't using "windows" to measure and check for stability.

Reply:
```suggestion
    // We only need to clamp the window when --request-count is set,
    // since we are no longer using fixed time windows.
    measure_config.clamp_window = (request_count != 0);
```
Would you be able to add something to this effect as a comment? It wouldn't be immediately clear to me looking at this code why this conditional works without needing to ask you or go back up the call stack to figure it out. Maybe something like the above?

Long-term refactoring comment (beyond the scope of this ticket): we should probably make the relationship between request_count and clamp_window more explicit/readable from the code itself, plus test for this behavior. If we end up using request_count differently in the future, we'll want to know to update these conditionals, so having them in one centralized place would be better.

Reply:
Done.

the first step of refactoring should probably be to avoid having 20(!) args passed in to the constructor of InferenceProfiler. There is just so much coupling and each class does wayyy more than one thing.

I tried to put this comment and the rest of the configuration logic into a constructor of MeasureConfig so that it is in one place and hidden away from the rest of the logic. I wanted to remove things like measurement_window_ms_ from being member variables that are captured in constructor and only used in one place to populate the MeasureConfig, but not quite everything was there and I don't have time to do a larger refactor.

Reply:
Perfect, thanks Tim!

That's excellent context. I'll try to incorporate it once we start the refactor work.

CC: @debermudez @matthewkotila @nv-hwoo so others have this context as well.